torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -309,6 +315,23 @@ def aot_dispatch_autograd(
             if not hasattr(compiled_fw_func, "_boxed_call"):
                 compiled_fw_func = make_boxed_func(compiled_fw_func)
 
+            if tracing_context and tracing_context.fakify_first_call:
+                out = [n.meta["val"] for n in (list(fw_module.graph.nodes)[-1].args[0])]
+                # will only be set for inductor
+                if fwd_output_strides:
+                    for i in range(len(out)):
+                        if not isinstance(out[i], Tensor):
+                            continue
+                        if all(
+                            guard_size_oblivious(s1 == s2)
+                            for s1, s2 in zip(out[i].stride(), fwd_output_strides[i])
+                        ):
+                            continue
+                        out[i] = out[i].as_strided(out[i].shape, fwd_output_strides[i])
+
+                fakified_out = out
```

Comment:
Hmm I'm actually a bit confused - if we're grabbing the FakeTensor graph outputs (with the proper inductor strides) from here, why do we need to bother invoking the AOTAutograd runtime code at all in order to send this info to ddp_optimizer? Why not stash this immediately in the tracing context and grab it?

I'm mostly curious because I'm wondering if I misunderstood something. It seems nice to avoid the "return fake tensors at runtime" situation if we can avoid it, since it does feel a bit like a footgun (we'd better make sure we never actually hit that codepath from when the user calls the runtime code).

Reply:
This is what I did in the first PR. My understanding is these outputs from inductor have no easy mapping to what is returned in dynamo.

Reply:
curious: with changes in this PR, is the output of first iteration going to be fake tensors when DDPOptimizer is enabled?

Reply:
The first invocation will be, which occurs in the graph splitter